pyccel/parser/syntactic.py

```diff
@@ -853,6 +853,13 @@ def _visit_ClassDef(self, stmt):
                 methods.append(self._visit(i))
             elif isinstance(i, ast.Pass):
                 return errors.report(UNSUPPORTED_FEATURE_OOP_EMPTY_CLASS, symbol = stmt, severity='error')
+        if not any(method.name == '__del__' for method in methods):
+            header = hdr_parse(stmts=f"#$ header method __del__({name})")
+            self.scope.parent_scope.insert_header(header)
+            arguments = ast.arguments(args=[ast.arg(arg='self')], kwonlyargs=[], defaults=[])
+            body = [Pass()]
+            del_functiondef = ast.FunctionDef(name="__del__", args=arguments, body=body, decorator_list=[])
+            methods.append(self._visit(del_functiondef))
         for i in methods:
```

Comment:
I don't suppose you can think of a way to write this in the semantic stage? Or avoid the header? I am working to deprecate these headers (see #1487 , #1523) so I don't want to introduce new ones in parallel.

Reply:
Fixed 13152019b5509fd4eb35fde7c5f9dff374192d56.